src/systems/battery_plugin/LinearBatteryPlugin.cc

```diff
@@ -506,14 +513,16 @@ double LinearBatteryPlugin::OnUpdateVoltage(
   this->dataPtr->ismooth = this->dataPtr->ismooth + k *
```

Comment:
Done in 5e00098.

Reply:
So `tau` has a range `[dt, +inf)`.

`k` has a range `[0, 1]` (`(0, 1]` strictly speaking) because `k = dt/tau`, but the user specifies `tau` not `k`.

Reply:
Oh, I see...Can we know the value of `dt` cat `Configure()` time?

Reply:
I don't think we can. We can just check that if `tau > 0` in `Configure` and clip it to `dt` if it's less than `dt` in `PreUpdate` with a warning message.

Reply:
I added the extra check in the `Update()` just because `stepSize` was already there. I think it should be fine too. f7fa498.